src/components/enterprise-user-subsidy/data/hooks.jsx

```diff
@@ -11,39 +11,120 @@ import { LICENSE_STATUS } from './constants';
+  } catch (error) {
+    logError(error);
+    return null;
+  }
+};
+
+/**
+ * Attempts to auto-apply a license for the authenticated user and the specified customer agreement.
+ *
+ * @param {string} customerAgreementId The UUID of the customer agreement.
+ * @returns An object representing the auto-applied license or null if no license was auto-applied.
+ */
+const requestAutoAppliedUserLicense = async (customerAgreementId) => {
+  try {
+    const response = await requestAutoAppliedLicense(customerAgreementId);
+    const license = camelCaseObject(response.data);
+    return license;
+  } catch (error) {
+    logError(error);
+    return null;
+  }
+};
+
+/**
+ * Retrieves a license for the authenticated user, if applicable. First attempts to find any existing licenses
+ * for the user. If a license is found, the app uses it; otherwise, if the enterprise has an SSO/LMS identity
+ * provider configured and the customer agreement has a subscription plan suitable for auto-applied licenses,
+ * attempt to auto-apply a license for the user.
+ *
+ * @param {object} args
+ * @param {object} args.enterpriseConfig The enterprise customer config
+ * @param {object} args.customerAgreementConfig The customer agreement config associated with the enterprise
+ * @param {boolean} args.isLoadingCustomerAgreementConfig Whether the customer agreement is still resolving
+ * @returns Array containing a user license, if applicable, and whether the license data is still resolving
+ */
+export function useSubscriptionLicense({
+  enterpriseConfig,
+  customerAgreementConfig,
+  isLoadingCustomerAgreementConfig,
+}) {
   const [license, setLicense] = useState();
   const [isLoading, setIsLoading] = useState(true);
 
+  const {
+    uuid: enterpriseId,
+    identityProvider: enterpriseIdentityProvider,
+  } = enterpriseConfig;
+
   useEffect(() => {
-    fetchSubscriptionLicensesForUser(enterpriseId)
-      .then((response) => {
-        const { results } = camelCaseObject(response.data);
-        const activated = results.filter(result => result.status === LICENSE_STATUS.ACTIVATED);
-        const assigned = results.filter(result => result.status === LICENSE_STATUS.ASSIGNED);
-        const revoked = results.filter(result => result.status === LICENSE_STATUS.REVOKED);
-
-        if (activated.length) {
-          setLicense(activated.shift());
-        } else if (assigned.length) {
-          setLicense(assigned.shift());
-        } else if (revoked.length) {
-          setLicense(revoked.shift());
-        } else {
-          setLicense(null);
+    async function retrieveUserLicense() {
+      let result = await fetchExistingUserLicense(enterpriseId);
+
+      if (features.ENABLE_AUTO_APPLIED_LICENSES) {
+        const customerAgreementMetadata = [
+          customerAgreementConfig?.uuid,
+          customerAgreementConfig?.subscriptionForAutoAppliedLicenses,
+        ];
+        const hasCustomerAgreementData = customerAgreementMetadata.every(item => !!item);
```

Comment:
nit: I think this might be a little easier to understand and less code:  
```
const hasCustomerAgreementData = customerAgreementConfig?.uuid && customerAgreementConfig?.subscriptionForAutoAppliedLicenses
```

Reply:
@manny-m 

This would result in `hasCustomerAgreementData` as the `customerAgreementConfig?.subscriptionForAutoAppliedLicenses` value or if `customerAgreementConfig` isn't defined yet (i.e., still loading), the value would returned as undefined:

![image](https://user-images.githubusercontent.com/2828721/142662665-051c544b-db4e-462f-8122-e521352e4ad8.png)

While this would technically work, I think it makes sense for this variable to be a true boolean value 😸